test/integration/puppetlabs/puppetserver/testutils.clj

```diff
@@ -97,8 +99,12 @@
   [puppet-conf-files :- PuppetConfFiles
    target-paths :- PuppetConfFiles]
   (doseq [filename (keys puppet-conf-files)]
-    (fs/copy+ (get puppet-conf-files filename)
-              (get target-paths filename))))
+    (let [dest (io/file (get target-paths filename))
+          dest-dir (.getParentFile dest)]
+      (when (and dest-dir (not (.exists dest-dir)))
+        (.mkdirs dest-dir))
```

Comment:
Could do this instead to eliminate a couple of Java method calls and have the function blow up if the directory creation fails for some reason:

```clj
(when dest-dir
  (ks/mkdirs! dest-dir)
```